QuickFIXn/Message/Message.cs

```diff
@@ -681,7 +681,7 @@ public void ReverseRoute(Header header)
                 this.Header.RemoveField(Tags.OnBehalfOfLocationID);
                 this.Header.RemoveField(Tags.DeliverToLocationID);
 
-                if (beginString.CompareTo("FIX.4.1") >= 0)
+                if (string.CompareOrdinal(beginString, "FIX.4.1") >= 0)
```

Comment:
As recommended by Microsoft - it's better to use Equals method to test equality, isn't it ?

Reply:
You're right, but here we are not checking for equality (of FIX version 4.1), rather that we are at version 4.1 _or higher_ (from the `>=`).